cmd/kubemark/hollow-node.go

```diff
@@ -86,6 +97,12 @@ func (c *hollowNodeConfig) addFlags(fs *pflag.FlagSet) {
 	fs.BoolVar(&c.UseRealProxier, "use-real-proxier", true, "Set to true if you want to use real proxier inside hollow-proxy.")
 	fs.DurationVar(&c.ProxierSyncPeriod, "proxier-sync-period", 30*time.Second, "Period that proxy rules are refreshed in hollow-proxy.")
 	fs.DurationVar(&c.ProxierMinSyncPeriod, "proxier-min-sync-period", 0, "Minimum period that proxy rules are refreshed in hollow-proxy.")
+
+	// for auth provider
+	fs.StringVar(&c.AuthProvider, "auth-provider", "", "The name of the auth provider (currently only supports 'eks').")
```

Comment:
I actually think that auth-provider will have completely different configuration for every setup.
So I would prefer moving those flags to a separate package and prefix all the flags with eks.

If really needed, we can converge it in the future.

Reply:
While I fully agree that auth-provider setting for EKS should be moved to cloud-provider specific code, I think the flags auth-provider-name and auth-provider-config should still live here because they're extension points for the kubeconfig itself (which of course can be set differently by different providers):

```
			clientConfig = &restclient.Config{
				Host: v.Server,
				TLSClientConfig: restclient.TLSClientConfig{},
				AuthProvider: &clientcmdapi.AuthProviderConfig{
					Name: config.AuthProvider,
					Config: map[string]string{
						blabla..
					},
				},
			}
```